unity-renderer/Assets/Scripts/MainScripts/DCL/Components/AvatarAttach/AvatarAttachHandler.cs

```diff
@@ -0,0 +1,159 @@
+
+        private Vector2Int? currentCords = null;
+        private bool isInsideScene = true;
+        private float lastBoundariesCheckTime = 0;
+
+        public void Initialize(IParcelScene scene, IDCLEntity entity)
+        {
+            this.scene = scene;
+            this.entity = entity;
+            getAnchorPointsHandler.OnAvatarRemoved += Detach;
+        }
+
+        public void OnModelUpdated(string json)
+        {
+            OnModelUpdated(model.GetDataFromJSON(json) as AvatarAttachComponent.Model);
+        }
+
+        public void OnModelUpdated(AvatarAttachComponent.Model newModel)
+        {
+            prevModel = model;
+            model = newModel;
+
+            if (model == null)
+            {
+                return;
+            }
+
+            if (prevModel.avatarId != model.avatarId)
+            {
+                Detach();
+
+                if (!string.IsNullOrEmpty(model.avatarId))
+                {
+                    Attach(model.avatarId, (AvatarAnchorPointIds)model.anchorPointId);
+                }
+            }
+        }
+
+        public void CleanUp()
+        {
+            Detach();
+            getAnchorPointsHandler.OnAvatarRemoved -= Detach;
+            getAnchorPointsHandler.Dispose();
+        }
+
+        internal virtual void Detach()
+        {
+            if (componentUpdate != null)
+            {
+                CoroutineStarter.Stop(componentUpdate);
+                componentUpdate = null;
+            }
+
+            if (entity != null)
+            {
+                entity.gameObject.transform.localPosition = EnvironmentSettings.MORDOR;
+            }
+
+            getAnchorPointsHandler.CancelCurrentSearch();
+        }
+
+        internal virtual void Attach(string avatarId, AvatarAnchorPointIds anchorPointId)
+        {
+            getAnchorPointsHandler.SearchAnchorPoints(avatarId, anchorPoints =>
+            {
+                Attach(anchorPoints, anchorPointId);
+            });
+        }
+
+        internal virtual void Attach(IAvatarAnchorPoints anchorPoints, AvatarAnchorPointIds anchorPointId)
+        {
+            this.anchorPoints = anchorPoints;
+            this.anchorPointId = anchorPointId;
+
+            if (componentUpdate == null)
+            {
+                componentUpdate = CoroutineStarter.Start(ComponentUpdate());
+            }
+        }
+
+        IEnumerator ComponentUpdate()
+        {
+            currentCords = null;
+
+            while (true)
+            {
+                if (entity == null || scene == null)
+                {
+                    componentUpdate = null;
+                    yield break;
+                }
+
+                var anchorPoint = anchorPoints.GetTransform(anchorPointId);
+
+                if (IsInsideScene(CommonScriptableObjects.worldOffset + anchorPoint.position))
+                {
+                    entity.gameObject.transform.position = anchorPoint.position;
```

Comment:
Just a warning: the timing here *might* be off if we don't use `LateUpdate`, this meaning that the attachments can be off by a frame respecting the avatar animations. 

At the same time `new WaitForEndOfFrame()` will not work on CI, so if this becomes an issue, we can use the `IUpdateEventHandler` that can be accessed by doing `Environment.i.platform.updateEventHandler`.